core/src/main/java/org/jboss/pnc/build/finder/protobuf/ListKojiArchiveInfoProtobufWrapper.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.build.finder.protobuf;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.infinispan.protostream.annotations.ProtoFactory;
+import org.infinispan.protostream.annotations.ProtoField;
+
+import com.redhat.red.build.koji.model.xmlrpc.KojiArchiveInfo;
+
+/**
+ * Class to wrap around a List of {@link com.redhat.red.build.koji.model.xmlrpc.KojiArchiveInfo}. This is used so that
+ * Protostream can properly marshall/unmarshall the list and avoid this Protostream bug
+ * (<a href="https://issues.redhat.com/browse/IPROTO-219">IPROTO-219</a>).
+ *
+ * @see <a href="https://issues.redhat.com/browse/IPROTO-219">IPROTO-219</a>
+ */
+public class ListKojiArchiveInfoProtobufWrapper {
+    /**
+     * We cannot use Collections.emptyList because Protostream has no idea how to marshall it. Instead, we use an empty
+     * ArrayList.
+     */
+    private static final List<KojiArchiveInfo> EMPTY_LIST = new ArrayList<>(0);
```

Comment:
Should/can we wrap this in `Collections.unmodifiableList()`?

Reply:
I don't think so. Protobuf doesn't know how to serialize those.

Reply:
Guess I'm wrong again

Reply:
Well, I think `Collections.singletonList()` is also unmodifiable, so if we can't use that. then `Collections.unmodifiableList()` seems like the next best thing. And the fact that protobuf can't serialize it is not your fault or even protobuf's fault really---I think it would need a separate adapter.